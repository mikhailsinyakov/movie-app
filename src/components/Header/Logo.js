import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

const HyperLink = ({ children, href, className }) => {
  if (href[0] === "/")
    return (
      <Link to={href} className={className}>
        {children}
      </Link>
    );
  else
    return (
      <a href={href} className={className}>
        {children}
      </a>
    );
};

const StyledHyperLink = styled(HyperLink)`
  height: var(--header-height);
  width: var(--header-height);
`;

const Logo = ({ filename, href, size, className }) => {
  const logoSrc = `img/${filename}`;
  const [name] = filename.split(".");
  return (
    <StyledHyperLink href={href}>
      <img src={logoSrc} alt={name} className={className} />
    </StyledHyperLink>
  );
};

Logo.propTypes = {
  filename: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  size: PropTypes.string
};

const StyledLogo = styled(Logo)`
  height: ${props => (props.size === "medium" ? "80%" : "50%")};
  margin: ${props => (props.size === "medium" ? "10%" : "25%")};
`;

export default StyledLogo;
