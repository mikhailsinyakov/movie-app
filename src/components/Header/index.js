import React from "react";
import styled from "styled-components";
import { useTranslation } from "react-i18next";
import Logo from "./Logo";

const Head = styled.h1`
  font-size: 1.5em;
  line-height: 1.5em;
  color: #c9fdb0;
  text-shadow: 0 0 5px black;
`;

const Header = ({ className }) => {
  const { t } = useTranslation();
  return (
    <header className={className}>
      <Logo filename="app-logo.svg" href="/" size="medium" />
      <Head>{t("appName")}</Head>
      <Logo
        filename="tmdb-logo.png"
        href="https://www.themoviedb.org/"
        size="small"
      />
    </header>
  );
};

const StyledHeader = styled(Header)`
  display: flex;
  justify-content: space-between;
  height: var(--header-height);
  background-image: linear-gradient(rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.7)),
    url(/img/tickets.jpg);
  background-position: center;
  background-size: cover;
  padding: 0.5rem;
  border-radius: 0.5rem;
`;

export default StyledHeader;
