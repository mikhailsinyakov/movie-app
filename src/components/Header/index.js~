import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Logo from "./Logo";
import { withLanguageContext } from "../Language";

const Head = styled.h1`
  font-size: 1.5em;
  line-height: 1.5em;
  color: #c9fdb0;
  text-shadow: 0 0 5px black;
`;

const Header = ({ getUIText, className }) => {
  return (
    <header className={className}>
      <Logo filename="app-logo.svg" href="/" size="medium" />
      <Head>{getUIText("appName")}</Head>
      <Logo
        filename="tmdb-logo.png"
        href="https://www.themoviedb.org/"
        size="small"
      />
    </header>
  );
};

Header.propTypes = {
  language: PropTypes.string.isRequired,
  setLanguage: PropTypes.func.isRequired,
  getUIText: PropTypes.func.isRequired
};

const StyledHeader = styled(Header)`
  display: flex;
  justify-content: space-between;
  height: var(--header-height);
  background-image: linear-gradient(rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.7)),
    url(/img/tickets.jpg);
  background-position: center;
  background-size: cover;
  padding: 0.5rem;
  border-radius: 0.5rem;
`;

export default withLanguageContext(StyledHeader);
