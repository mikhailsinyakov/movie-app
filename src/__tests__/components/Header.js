import React from "react";
import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import withLanguage from "../../components/Language";
import Header from "../../components/Header";

const Parent = ({ children }) => <div>{children}</div>;
const ParentWithLanguage = withLanguage(Parent);

let component, instance;
beforeAll(() => {
  component = create(
    <Router>
      <ParentWithLanguage>
        <Header />
      </ParentWithLanguage>
    </Router>
  );
  instance = component.root;
});

it("matches snapshot", () => {
  expect(component.toJSON()).toMatchSnapshot();
});

it("has two images", () => {
  const images = instance.findAllByType("img");
  expect(images).toHaveLength(2);
});

it("has a header text", () => {
  const headerText = instance.findByType("h1");
  expect(headerText).toBeTruthy();
});
