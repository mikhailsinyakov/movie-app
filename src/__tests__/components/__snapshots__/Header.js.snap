// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="app-logo"
        src="img/app-logo"
      />
    </a>
    <h1>
      Movie App
    </h1>
    <select
      onChange={[Function]}
      value="en"
    >
      <option
        value="en"
      >
        EN
      </option>
      <option
        value="ru"
      >
        RU
      </option>
    </select>
    <a
      href="https://www.themoviedb.org/"
    >
      <img
        alt="tmdb-logo"
        src="img/tmdb-logo"
      />
    </a>
  </header>
</div>
`;
